src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -232,6 +230,34 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                             rawResource = await streamReader.ReadToEndAsync();
                         }
 
+                        // if we have more columns, it means sort expressions were added.
+                        if (reader.FieldCount > 9)
+                        {
+                            try
+                            {
+                                // if sort is requested asc keep the first result for continuation token
+                                if (matchCount == 0 && searchOptions.Sort?[0].sortOrder == SortOrder.Ascending)
+                                {
+                                    sortExpr = reader.GetValue("SortExpr") as DateTime?;
+                                }
+                                else if (matchCount > 0 && searchOptions.Sort?[0].sortOrder == SortOrder.Descending)
+                                {
+                                    // Otherwise keep getting this, until we got the last result for Desc
+                                    sortExpr = reader.GetValue("SortExpr") as DateTime?;
+                                }
+                            }
+                            catch
```

Comment:
Please don't add random swallows like this.

Reply:
Done